pkg/api/vlabs/types.go

```diff
@@ -162,6 +162,8 @@ type CustomNodesDNS struct {
 type WindowsProfile struct {
 	AdminUsername          string            `json:"adminUsername,omitempty"`
 	AdminPassword          string            `json:"adminPassword,omitempty"`
+	CSIProxyURL            string            `json:"csiProxyURL"`
```

Comment:
I think we should `omitempty` here. I've also noticed other properties are probably missing this, and filed a backlog issue here:

https://github.com/Azure/aks-engine/issues/2933

Reply:
@jackfrancis should I add omitempty only for vlabs/types.go or pkg/api/types.go as well?

Reply:
to be consistent and cautious I always duplicate it in both versions, tho AFAIK the vlabs is the only one that matters in terms of user api model impact

Reply:
k - will update

Reply:
updated